src/tbb/misc_ex.cpp

```diff
@@ -297,11 +297,21 @@ static void initialize_hardware_concurrency_info () {
         if ( pam & m )
             ++nproc;
     }
-    __TBB_ASSERT( nproc <= (int)si.dwNumberOfProcessors, nullptr);
+    int number_of_processors = (int)si.dwNumberOfProcessors;
+    if (nproc > number_of_processors && TBB_GetThreadGroupAffinity) {
+        // Sometimes on systems with multiple processor groups GetNativeSystemInfo
+        // reports mask and processor count from the parent process
+        TBB_GROUP_AFFINITY ga;
+        if (TBB_GetThreadGroupAffinity(GetCurrentThread(), &ga)) {
+            number_of_processors = (int)TBB_GetActiveProcessorCount(ga.Group);
```

Comment:
1. Should we just always be using `GetActiveProcessorCount` if it is more reliable?
2. Just curious why we are checking for `TBB_GetThreadGroupAffinity` here but not checking for `TBB_GetActiveProcessorCount`?

Reply:
1. I am not sure about that. The code seems to be designed in a way that these Group-related features may not be available. So I decided not to replace that functionality, but rather complement it.
2. It is actually enough to check just for one symbol. `dynamic_link` module in TBB works in a way that if some of the symbols in symbol table is not found, than the whole DLL won't be loaded and thus other symbols will be unavailable as well.